front/pages/w/[wId]/assistant/new.tsx

```diff
@@ -74,66 +67,51 @@ export const getServerSideProps = withDefaultUserAuthRequirements<
       owner,
       subscription,
       user,
+      plan,
     },
   };
 });
 
 export default function AssistantNew({
-  helper,
-  isBuilder,
   owner,
   subscription,
   user,
 }: InferGetServerSidePropsType<typeof getServerSideProps>) {
   const router = useRouter();
+
   const [planLimitReached, setPlanLimitReached] = useState<boolean>(false);
   const sendNotification = useContext(SendNotificationsContext);
-  const [conversationHelperModal, setConversationHelperModal] =
-    useState<ConversationType | null>(null);
 
-  // No limit on global assistants call as they include both active and inactive.
-  const globalAgentConfigurations = useAgentConfigurations({
+  const { animate, setAnimate, setSelectedAssistant } =
+    useContext(InputBarContext);
+
+  const assistantToMention = useRef<LightAgentConfigurationType | null>(null);
+
+  // fast loading of a few company assistants so we can show them immediately
+  const {
+    agentConfigurations: initialAgentConfigurations,
+    isAgentConfigurationsLoading: isInitialAgentConfigurationsLoading,
+  } = useAgentConfigurations({
     workspaceId: owner.sId,
-    agentsGetView: "global",
-    includes: ["authors"],
-    sort: "priority",
-  }).agentConfigurations;
-
-  const workspaceAgentConfigurations = isBuilder
-    ? []
-    : useAgentConfigurations({
-        workspaceId: owner.sId,
-        agentsGetView: "workspace",
-        includes: ["authors"],
-        limit: 2,
-        sort: "alphabetical",
-      }).agentConfigurations;
-
-  const displayedAgents = [
-    ...globalAgentConfigurations,
-    ...workspaceAgentConfigurations,
-  ]
-    .filter((a) => a.status === "active")
-    // Sort is necessary due to separately fetched global and workspace assistants, ensuring unified ordering.
-    .sort(compareAgentsForSort)
-    .slice(0, isBuilder ? 2 : 4);
+    agentsGetView: "workspace",
+    limit: 24,
+  });
 
+  // we load all assistants with authors in the background
   const {
```

Comment:
OOC, any reason to keep these hooks here rather than in the `AssistantBrowser`?

Reply:
AssistantBrowser is a purer UX component that  way, just in charge of rendering, not in charge of knowing the intricate logic loading (mostly style)

Reply:
Yeah not fully convinced but it's a way better than the first work \o/. I'll live with it.